File: 프로그래머스/1/12982. 예산/예산.java
import java.util.*;

class Solution {
    public int solution(int[] d, int budget) {
        int ans = 0;
        Arrays.sort(d);
        
        for (int i : d) {
            if (budget - i < 0) {
                return ans;
            }
            
            budget -= i;
            ans++;
        }
        return d.length;
    }
}

File: 프로그래머스/1/12982. 예산/예산.js
function solution(d, budget) {
    d.sort((a, b) => a - b);

    let total = 0;
    let count = 0;
    
    for (let i = 0; i < d.length; i++) {
        total += d[i];
        if (total > budget) {
            break;
        }
        count++;
    }
    
    return count;
}
